src/libraries/Microsoft.Extensions.Diagnostics/src/DefaultMeterFactory.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+namespace Microsoft.Extensions.Metrics
+{
+    internal sealed class DefaultMeterFactory : IMeterFactory
+    {
+        private readonly Dictionary<string, List<Meter>> _cachedMeters = new();
+        private bool _disposed;
+
+        public DefaultMeterFactory() { }
+
+        public Meter Create(MeterOptions options)
+        {
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            Debug.Assert(options.Name != null);
+
+            lock (_cachedMeters)
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException(GetType().Name);
+                }
+
+                if (_cachedMeters.TryGetValue(options.Name, out List<Meter>? meterList))
+                {
+                    foreach (Meter meter in meterList)
+                    {
+                        if (meter.Version == options.Version && CompareTags(meter.Tags, options.Tags))
+                        {
+                            return meter;
+                        }
+                    }
+                }
+                else
+                {
+                    meterList = new List<Meter>();
+                    _cachedMeters.Add(options.Name, meterList);
+                }
+
+                Meter m = new Meter(options.Name, options.Version, options.Tags, scope: this);
```

Comment:
Weren't we going to use a subclassed Meter who's exposed Dispose was a nop?

Reply:
I am going to track the disposal issue to look more at the best solution we could go with.

Reply:
Ok.  If we don't end up changing this, we need to change the name of the method.  Calling it Create and returning something that's IDisposable but then telling people in docs not to Dispose it will lead to a pit of failure.

Reply:
I opened https://github.com/dotnet/runtime/issues/86592